src/NUnitFramework/framework/Internal/Tests/TestSuite.cs

```diff
@@ -117,7 +117,9 @@ public TestSuite(TestSuite suite, ITestFilter filter)
                 {
                     if(child.IsSuite)
                     {
-                        TestSuite childSuite = new TestSuite(child as TestSuite, filter);
+                        TestSuite childSuite = child.GetType() == typeof(TestFixture)
```

Comment:
Personally, I think we should go for a more general solution for all classes deriving from `TestSuite` and not only handle `TestFixture` as a special case. We could introduce a copy constructor for each deriving class and also an instance method that returns an instance of the same type as the instance, e.g. in `TestFixture`
```c#
public void Copy(TestSuite suite, ITestFilter filter)
{
    return new TestFixture(suite, filter);
}
```
then we could replace the code with
```c#
TestSuite childSuite = child.Copy(child as TestSuite, filter);
```

Reply:
I agree. We should do whatever it takes to avoid a type test.

Reply:
Makes sense. I've updated with something similar to @mikkelbu's proposal.